coralproject/talk
# Asset Performance Fix

## What does this PR do?

- Fixes #2158
  - Adds new indexes to assets to improve query performance
  - Adds new limits to the query sizes of the assets API edge to improve performance
  - Cleaned up some CSS when dealing with large page numbers

## How do I test this PR?

1. Load assets into your database (assuming you're running this on port 3000 in development mode):

```js
let assets = [];
for (var i = 0; i < 75000; i++) {
  assets.push({url:"http://localhost:3000/dev/assets/title/" + i, closedAt:null,closedMessage:null,created_at: new ISODate(), id: "" + i + "", metadata: {}, scraped: null, settings: {}, tags: [], type: "assets", updated_at: new ISODate()});
}
db.assets.insert(assets);
```

2. Load the settings page, notice that the page loads! (Note, there will be lots of missing data there, title, publication date, but that's ok!)
